pkg/bgpv1/manager/reconcilerv2/neighbor.go

```diff
@@ -264,6 +305,76 @@ func (r *NeighborReconciler) Reconcile(ctx context.Context, p ReconcileParams) e
 	return nil
 }
 
+// getDefaultGateway returns the default gateway address with lower priority using route and device
+// statedb tables and the provided default gateway configuration.
+func (r *NeighborReconciler) getDefaultGateway(defaultGateway *v2.DefaultGateway) (string, error) {
+	var defaultRoute string
+	switch defaultGateway.AddressFamily {
+	case "ipv4":
+		defaultRoute = netip.PrefixFrom(netip.IPv4Unspecified(), 0).String()
+	case "ipv6":
+		defaultRoute = netip.PrefixFrom(netip.IPv6Unspecified(), 0).String()
+	default:
+		return "", fmt.Errorf("invalid address family %s", defaultGateway.AddressFamily)
+	}
+	// get routes from statedb route table
+	txn := r.DB.ReadTxn()
+	routeMeta := r.DB.GetTable(txn, "routes")
+	routeTbl := statedb.AnyTable{Meta: routeMeta}
+	routeObjs := routeTbl.All(txn)
+	header := routeTbl.TableHeader()
+	// extract indexes of required columns
+	routeColumns := []string{"Destination", "Source", "Gateway", "LinkIndex", "Priority"}
+	routeIdxs, err := getColumnIndexes(routeColumns, header)
+	if err != nil {
+		return "", fmt.Errorf("failed to get column indexes for route table: %w", err)
+	}
+
+	// get links from statedb device table
+	deviceMeta := r.DB.GetTable(txn, "devices")
+	deviceTbl := statedb.AnyTable{Meta: deviceMeta}
+	deviceObjs := deviceTbl.All(txn)
+	deviceHeader := deviceTbl.TableHeader()
+	// extract indexes of required columns
+	deviceColumns := []string{"Index", "OperStatus"}
+	deviceIdxs, err := getColumnIndexes(deviceColumns, deviceHeader)
+	if err != nil {
+		return "", fmt.Errorf("failed to get column indexes for device table: %w", err)
+	}
```

Comment:
Please don't list from statedb in this generic way, referencing to column names. Instead, list devices/routes like this to get typed `*tables.Device` / `*tables.Route`:

`deviceTable` is `statedb.Table[*tables.Device]`:

```golang
	devices := deviceTable.All(txn)
	for dev := range devices {
		fmt.Printf("device name is %s\n", dev.Name)
	}
```

(use `Get()` or `List()` methods to lookup by a table index)

Reply:
You are a SAVIOR! thank you for letting me know. i was referring the code from cilium-cli and i could not figure out how i can get typed data.
this is fixed!